applications/launchpad/backend/src/docker/workspace.rs

```diff
@@ -466,44 +457,32 @@ impl TariWorkspace {
 
     /// Stop the container with the given `name` and optionally delete it
     pub async fn stop_container(&mut self, name: &str, delete: bool, docker: &Docker) {
-        let container = match self.container_mut(name) {
-            Some(c) => c,
-            None => {
-                info!(
-                    "Cannot stop container {}. It was not found in the current workspace",
-                    name
-                );
-                return;
-            },
-        };
-        let options = StopContainerOptions { t: 0 };
-        let id = container.id().clone();
-        match docker.stop_container(id.as_str(), Some(options)).await {
-            Ok(_res) => {
-                info!("Container {} stopped", id);
-                container.set_stop();
-            },
-            Err(err) => {
-                warn!("Could not stop container {} due to {}", id, err.to_string());
-            },
-        }
-        // Even if stopping failed (maybe it was already stopped), try and delete it
-        if delete {
-            match docker.remove_container(id.as_str(), None).await {
-                Ok(()) => {
-                    info!("Container {} deleted", id);
-                    container.set_deleted();
+        let about_to_delete = CONTAINERS.write().unwrap().remove(name);
```

Comment:
There are problems with this rewrite. The original logic is better IMO. 
* Removing the state record is not the same as deleting the container
* Don't remove the container state object from the hashmap. Mark it as deleted. The idea here is to keep a record of deleted containers so that the front-end can differentiate between "destroyed" and "never existed".
* Update the fields if just stopping the container rather than removing and re-inserting. Doing this uses multiple locks and sort of defeats the point of having a "state" record in the first place.